tools/cmd/api-generator/convert_commands.go

```diff
@@ -419,54 +418,88 @@ func processRequestBody(requestBody *openapi3.RequestBodyRef, versionsMap map[st
+		if err := addContentTypeToVersion(versionedContentType, versionsMap, mediaType.Extensions, true); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
 // Helper function to add content type to version map.
-func addContentTypeToVersion(versionedContentType string, versionsMap map[string]*api.Version, isRequest bool, sunset *time.Time) error {
+func addContentTypeToVersion(versionedContentType string, versionsMap map[string]*api.CommandVersion, extensions map[string]any, isRequest bool) error {
+	// Extract the version and content type from the versioned content type.
 	version, contentType, err := extractVersionAndContentType(versionedContentType)
 	if err != nil {
 		return fmt.Errorf("unsupported version %q error: %w", versionedContentType, err)
 	}
 
-	if shouldIgnoreVersion(version) {
-		return nil
-	}
+	// Extract the sunset date and private preview from the extensions.
+	sunset := extractSunsetDate(extensions)
+	publicPreview := extractPublicPreview(extensions)
 
-	if _, ok := versionsMap[version]; !ok {
-		versionsMap[version] = &api.Version{
+	// Add the version to the versions map if it doesn't exist.
+	versionString := version.ToString()
+	if _, ok := versionsMap[versionString]; !ok {
+		versionsMap[versionString] = &api.CommandVersion{
 			Version:              version,
 			Sunset:               sunset,
 			ResponseContentTypes: []string{},
 		}
 	}
 
+	// If the sunset date is set, update the sunset date if it's before the current sunset date.
 	if sunset != nil {
-		if versionsMap[version].Sunset == nil || sunset.Before(*versionsMap[version].Sunset) {
-			versionsMap[version].Sunset = sunset
+		if versionsMap[versionString].Sunset == nil || sunset.Before(*versionsMap[versionString].Sunset) {
+			versionsMap[versionString].Sunset = sunset
 		}
 	}
 
+	// The default for public preview is false, override it if the extension says we're in a public preview.
+	if publicPreview != nil && *publicPreview {
+		versionsMap[versionString].PublicPreview = true
+	}
+
+	// If the versioned content type is a request, set the request content type.
+	// If the versioned content type is a response, add the content type to the response content types.
 	if isRequest {
-		if versionsMap[version].RequestContentType != "" {
+		if versionsMap[versionString].RequestContentType != "" {
 			return errors.New("multiple request content types is not supported")
 		}
 
-		versionsMap[version].RequestContentType = contentType
+		versionsMap[versionString].RequestContentType = contentType
 	} else {
-		versionsMap[version].ResponseContentTypes = append(versionsMap[version].ResponseContentTypes, contentType)
+		versionsMap[versionString].ResponseContentTypes = append(versionsMap[versionString].ResponseContentTypes, contentType)
+	}
+
+	return nil
+}
+
+// Extract public preview from extensions.
+// Example yaml:
+// ```yaml
+// x-xgen-preview:
+//
+//	name: charts-dashboards
```

Comment:
`name` is set only for private preview since all the public preview API are in the same file.